packages/core/src/dispatcher.ts

```diff
@@ -134,7 +141,21 @@ export default class Dispatcher extends EventEmitter {
    * @param cid - Record CID
    */
   async retrieveRecord (cid: CID | string): Promise<any> {
-    return cloneDeep((await this._ipfs.dag.get(cid)).value)
+    const record = await this._ipfs.dag.get(cid, { timeout: DEFAULT_GET_TIMEOUT })
```

Comment:
Will this work if `cid` is a string, or do we need to normalize it to be a CID object?

Reply:
Yeah, it works both ways: `CID` | `string`.